.circleci/config.yml

```
@@ -0,0 +1,64 @@
+version: 2
+
+aliases:
+  - &restore-cache
+    restore_cache:
+        key: dependency-cache-{{ checksum "package.json" }}
+  - &install-deps
+    run:
+        name: Install dependencies
+        command: npm ci
+  - &lint
+    run:
+        name: Lint
+        command: npm run lint
+
+base-template: &base-template
+  working_directory: ~/try
+  steps:
+    - checkout
+    - *restore-cache
+    - *install-deps
+    - *lint
+
+jobs:
+  build:
+    working_directory: ~/try
+    docker:
+      - image: circleci/node:12
+    steps:
+      - checkout
+      - run:
+          name: Update NPM version
+          command: 'sudo npm install -g npm@latest'
+      - restore_cache:
+          key: dependency-cache-{{ checksum "package.json" }}
```

```suggestion
          key: dependency-cache-{{ checksum "package-lock.json" }}
```
If you're using `npm ci` it makes more sense to cache by the lock file